pkg/services/hivelocity/device/device.go

```diff
@@ -145,79 +129,50 @@ func (s *Service) Reconcile(ctx context.Context) (_ *ctrl.Result, err error) {
 	return nil, nil
 }
 
-func (s *Service) handleDeviceStatusOff(ctx context.Context, device *hv.BareMetalDevice) (*reconcile.Result, error) {
-	// Check if device is in DeviceStatusOff and turn it on. This is to avoid a bug of Hivelocity where
-	// sometimes machines are created and not turned on
-
-	condition := conditions.Get(s.scope.HivelocityMachine, infrav1.DeviceReadyCondition)
-	if condition != nil &&
-		condition.Status == corev1.ConditionFalse &&
-		condition.Reason == infrav1.DeviceOffReason {
-		if time.Now().Before(condition.LastTransitionTime.Time.Add(deviceOffTimeout)) {
-			// Not yet timed out, try again to power on
-			if err := s.scope.HVClient.PowerOnDevice(ctx, device.DeviceId); err != nil {
-				return nil, fmt.Errorf("failed to power on device: %w", err)
-			}
-		} else {
-			// Timed out. Set failure reason
-			s.scope.SetError("reached timeout of waiting for machines that are switched off", capierrors.CreateMachineError)
-			return nil, nil
+func (s *Service) updateDevice(ctx context.Context, log logr.Logger, device *hv.BareMetalDevice) error {
+	log.V(1).Info("Updating machine")
+
+	// if the machine is already provisioned, return
+	if s.scope.Machine.Spec.ProviderID != nil {
+		// ensure ready state is set.
+		// This is required after move, because status is not moved to the target cluster.
+		s.scope.HivelocityMachine.Status.Ready = true
+		deviceID, err := hvutils.ProviderIDToDeviceID(*s.scope.Machine.Spec.ProviderID)
+		if err != nil {
+			return fmt.Errorf("[updateDevice] ProviderIDToDeviceID failed: %w", err)
 		}
-	} else { // todo: too complicated. Is there a way to avoid the "else"?
-		// No condition set yet. Try to power device on.
-		if err := s.scope.HVClient.PowerOnDevice(ctx, device.DeviceId); err != nil {
-			if hvclient.IsRateLimitExceededError(err) {
-				conditions.MarkTrue(s.scope.HivelocityMachine, infrav1.RateLimitExceeded)
-				record.Event(s.scope.HivelocityMachine,
-					"RateLimitExceeded",
-					"exceeded rate limit with calling hivelocity function PowerOnDevice",
-				)
-			}
-			return nil, fmt.Errorf("failed to power on device: %w", err)
+		exists, err := hvutils.DeviceExists(ctx, s.scope.HVClient, deviceID)
+		if err != nil {
+			return fmt.Errorf("[updateDevice] DeviceExists failed: %w", err)
 		}
-		conditions.MarkFalse(
-			s.scope.HivelocityMachine,
-			infrav1.DeviceReadyCondition,
-			infrav1.DeviceOffReason,
-			clusterv1.ConditionSeverityInfo,
-			"device is switched off",
-		)
+		if exists {
+			conditions.MarkTrue(s.scope.HivelocityMachine, infrav1.DeviceReadyCondition)
+			// Setting address is required after move, because Status field is not retained during move.
```

Comment:
I don't really like this comment here and the equivalent for status.ready. It makes it a bit "too" easy.. I would remove it altogether, because this step follows clearly the pattern that is given by the Cluster API (see the steps written down here https://cluster-api.sigs.k8s.io/developer/providers/machine-infrastructure.html). 
As this document (or the book in general) is referenced in the README, I wouldn't try to argue in the code why all of these steps are necessary.

Reply:
I removed the comment.